# Inicio Processo

### Libs

In [1]:
import pandas as pd
import numpy as np

# Importando dadaos & VIsualização de Tabelas

### Datasets

In [2]:
dataStoretInfo = pd.read_csv('docs_csv/data_store_info.csv')
dataStoreSales = pd.read_csv('docs_csv/data_store_sales.csv')
dataProductSales = pd.read_csv('docs_csv/data_product_sales.csv')
productInformation = pd.read_csv('docs_csv/data_product_info.csv')
IMDB_movies = pd.read_csv('docs_csv/IMDB_movies.csv')

### Visualizando dados

In [3]:
print(f'Quantidade de itens: {dataStoretInfo.shape}')
print('\n', dataStoretInfo.head())
print('\n_______________INFORMAÇÕES_______________')
print(dataStoretInfo.info())
print('\n_______________VALORES NULOS_______________')
print(dataStoretInfo.isnull().sum())

Quantidade de itens: (20, 6)

    STORE_CODE STORE_NAME  START_DATE  END_DATA BUSINESS_NAME  BUSINESS_CODE
0           1  Sao Paulo  2006-10-01       NaN        Varejo              1
1           2    Chicago  2007-10-01       NaN        Varejo              1
2           3       Roma  2008-10-01       NaN        Varejo              1
3           4      Tokio  2009-10-01       NaN        Varejo              1
4           5      Paris  2019-01-01       NaN   Proximidade              2

_______________INFORMAÇÕES_______________
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20 entries, 0 to 19
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   STORE_CODE     20 non-null     int64  
 1   STORE_NAME     20 non-null     object 
 2   START_DATE     20 non-null     object 
 3   END_DATA       0 non-null      float64
 4   BUSINESS_NAME  20 non-null     object 
 5   BUSINESS_CODE  20 non-null     int64  
dtypes: float64(

In [4]:
print(f'Quantidade de itens: {dataProductSales.shape}')
print('\n_______________INFORMAÇÕES_______________')
print(dataProductSales.info())
print('\n_______________VALORES NULOS_______________')
print(dataProductSales.isnull().sum())
dataProductSales.head()

Quantidade de itens: (2173133, 5)

_______________INFORMAÇÕES_______________
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2173133 entries, 0 to 2173132
Data columns (total 5 columns):
 #   Column        Dtype  
---  ------        -----  
 0   STORE_CODE    int64  
 1   PRODUCT_CODE  int64  
 2   DATE          object 
 3   SALES_VALUE   float64
 4   SALES_QTY     int64  
dtypes: float64(1), int64(3), object(1)
memory usage: 82.9+ MB
None

_______________VALORES NULOS_______________
STORE_CODE      0
PRODUCT_CODE    0
DATE            0
SALES_VALUE     0
SALES_QTY       0
dtype: int64


STORE_CODE  PRODUCT_CODE        DATE  SALES_VALUE  SALES_QTY
0           1            18  2019-01-01        708.5         65
1           1            18  2019-01-02       1297.1        119
2           1            18  2019-01-03       1144.5        105
3           1            18  2019-01-04       1090.0        100
4           1            18  2019-01-05        893.8         82

In [5]:
print(f'Quantidade de itens: {dataStoreSales.shape}')
print('\n_______________INFORMAÇÕES_______________')
print(dataStoreSales.info())
print('\n_______________VALORES NULOS_______________')
print(dataProductSales.isnull().sum())
dataStoreSales.head()

Quantidade de itens: (35762, 4)

_______________INFORMAÇÕES_______________
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 35762 entries, 0 to 35761
Data columns (total 4 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   STORE_CODE   35762 non-null  int64  
 1   DATE         35762 non-null  object 
 2   SALES_VALUE  35762 non-null  float64
 3   SALES_QTY    35762 non-null  int64  
dtypes: float64(1), int64(2), object(1)
memory usage: 1.1+ MB
None

_______________VALORES NULOS_______________
STORE_CODE      0
PRODUCT_CODE    0
DATE            0
SALES_VALUE     0
SALES_QTY       0
dtype: int64


STORE_CODE        DATE  SALES_VALUE  SALES_QTY
0           1  2014-11-23    111387.87       7227
1          10  2014-11-23     63256.35       4362
2          11  2014-11-23     62680.86       4257
3          12  2014-11-23     62680.86       4257
4          13  2014-11-23    111387.87       7227

In [6]:
print(f'Quantidade de itens: {productInformation.shape}')
print('\n_______________INFORMAÇÕES_______________')
print(productInformation.info())
print('\n_______________VALORES NULOS_______________')
print(productInformation.isnull().sum())
productInformation.head()

Quantidade de itens: (9994, 7)

_______________INFORMAÇÕES_______________
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 7 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   PRODUCT_COD   9994 non-null   int64  
 1   PRODUCT_NAME  9994 non-null   object 
 2   PRODUCT_VAL   9994 non-null   float64
 3   DEP_NAME      9994 non-null   object 
 4   DEP_COD       9994 non-null   int64  
 5   SECTION_NAME  9994 non-null   object 
 6   SECTION_COD   9994 non-null   int64  
dtypes: float64(1), int64(3), object(3)
memory usage: 546.7+ KB
None

_______________VALORES NULOS_______________
PRODUCT_COD     0
PRODUCT_NAME    0
PRODUCT_VAL     0
DEP_NAME        0
DEP_COD         0
SECTION_NAME    0
SECTION_COD     0
dtype: int64


PRODUCT_COD                                       PRODUCT_NAME  \
0           10  Acido Tranexamico 250mg Generico EMS 12 Compri...   
1           11  Bissulfato de Clopidogrel 75mg Generico Teuto ...   
2           12  Cloridrato de Amiodarona 200mg Generico Biosin...   
3           13  Acido Tranexâmico 250mg Generico Legrand 12 Co...   
4           14  Cloridrato Oximetazolina Adulto 5mg/ml Genéric...   

   PRODUCT_VAL                DEP_NAME  DEP_COD         SECTION_NAME  \
0        36.71  MEDICAMENTOS GENÉRICOS       10           CIRCULAÇÃO   
1        66.51  MEDICAMENTOS GENÉRICOS       10           CIRCULAÇÃO   
2        31.17  MEDICAMENTOS GENÉRICOS       10           CIRCULAÇÃO   
3        36.71  MEDICAMENTOS GENÉRICOS       10           CIRCULAÇÃO   
4        12.08  MEDICAMENTOS GENÉRICOS       10  GRIPES E RESFRIADOS   

   SECTION_COD  
0           40  
1           40  
2           40  
3           40  
4           41

# Perguntas

### <strong>Quantos produtos a empresa possui? </strong>

 > R: A empresa possui <strong>9962</strong> produtos
 
 <span class="mark">Observação</span> : Existem produtos com nomes duplicados, a diferença está apenas no código do produto e outros no preço.
    Se essa diferença não for significante, afirmo que existem <strong>9962</strong> produtos cadastrados, caso contrário considere um total de <strong>9994</strong> produtos.<br>Abaixo deixo o dataset apenas com os itens duplicados para averiguar.

#### linhas de código - pergunta 1

In [7]:
print(f"Quantidade de produtos: ",len(pd.unique(productInformation['PRODUCT_NAME'])))

Quantidade de produtos:  9962


In [8]:
print(f"Valores Nulos: ", productInformation['PRODUCT_NAME'].isnull().sum())
print(f"Produtos com nomes duplicados: ", productInformation['PRODUCT_NAME'].duplicated().sum())

Valores Nulos:  0
Produtos com nomes duplicados:  32


In [9]:
produtosDuplicados = productInformation.loc[productInformation['PRODUCT_NAME'].duplicated() == True]

In [10]:
lista = []

In [11]:
for i, v in enumerate(produtosDuplicados['PRODUCT_NAME']):
    lista.append(productInformation.loc[productInformation['PRODUCT_NAME'] == v].values)

In [12]:
df = pd.DataFrame(lista[0], columns = ['PRODUCT_COD','PRODUCT_NAME','PRODUCT_VAL','DEP_NAME', 'DEP_COD','SECTION_NAME','SECTION_COD'])

In [13]:
for i, v in enumerate(lista, start=1):
    try:
        contador = 0
        
        while True:
            if contador == 2:
                break
            else:
                df.loc[len(df)+1] = lista[i][contador]
                contador += 1
    except:
        print('Finalizado')

Finalizado


In [14]:
df['DOUBLE'] = df['PRODUCT_NAME'].duplicated()
print(df.shape)
df.head(50)

(64, 8)


PRODUCT_COD                                       PRODUCT_NAME PRODUCT_VAL  \
0        12508  Absorvente INTIMUS GEL Tripla Protecao sem Aba...        3.59   
1        13369  Absorvente INTIMUS GEL Tripla Protecao sem Aba...        3.59   
3        12619                Queijo Gouda SERRABELLA Pedaco 200g       16.94   
4        15787                Queijo Gouda SERRABELLA Pedaco 200g       19.06   
5        38218  Cereal Integral a Base de Aveia CROQUI Origina...        9.99   
6        38515  Cereal Integral a Base de Aveia CROQUI Origina...        10.5   
7        22330  Lustra Moveis Lavanda BRAVO Johnsons Frasco 500ml        9.55   
8        50160  Lustra Moveis Lavanda BRAVO Johnsons Frasco 500ml        8.19   
9        99550                        Abacaxi Hawai Super Unidade        7.69   
10      136399                        Abacaxi Hawai Super Unidade        5.59   
11       51912  Aparelho de Barbear GILLETTE Prestobarba Ultra...        8.99   
12      137888  Aparelho de Barbear GILLETTE Prestobarba Ultra...        8.99   
13      277222      Desodorante Roll On NIVEA Stress Protect 50ml        7.09   
14      277223      Desodorante Roll On NIVEA Stress Protect 50ml        7.09   
15      278880  Creme Dental COLGATE Sensitive Pro-Alivio Bran...       11.65   
16      279177  Creme Dental COLGATE Sensitive Pro-Alivio Bran...       11.65   
17      295368       Cachaca Brasilizar YPIOCA Ouro Garrafa 965ml        13.9   
18      295369       Cachaca Brasilizar YPIOCA Ouro Garrafa 965ml        21.9   
19       33025  Racao para Gatos Sabor Carne WHISKAS Pratos Fa...        1.65   
20      310989  Racao para Gatos Sabor Carne WHISKAS Pratos Fa...        1.65   
21       21730                        Salsicha Hot Dog SEARA 500g        6.79   
22      315838                        Salsicha Hot Dog SEARA 500g         3.7   
23       91384                Achocolatado Liquido TODDYNHO 200ml        1.59   
24      319352                Achocolatado Liquido TODDYNHO 200ml        2.85   
25       60253          Bala de Gelatina Tubinhos FINI Pacote 80g        4.09   
26      324949          Bala de Gelatina Tubinhos FINI Pacote 80g        3.99   
27      326534  Kit com 3 Caixas de Fraldas PAMPERS Premium Ca...       185.8   
28      326560  Kit com 3 Caixas de Fraldas PAMPERS Premium Ca...       278.7   
29      326655  Kit com 2 Fraldas HUGGIES Veste Facil Jumbo Gr...        43.5   
30      326718  Kit com 2 Fraldas HUGGIES Veste Facil Jumbo Gr...        43.5   
31      326711  Kit com 2 Fraldas TURMA DA MONICA Mega Extra G...       76.78   
32      326719  Kit com 2 Fraldas TURMA DA MONICA Mega Extra G...       76.78   
33      326713  Kit com 2 Fraldas HUGGIES Veste Facil Jumbo Ex...        43.5   
34      326720  Kit com 2 Fraldas HUGGIES Veste Facil Jumbo Ex...        43.5   
35      326730  Kit com 2 Fraldas TURMA DA MONICA Media Pacote...       60.98   
36      326732  Kit com 2 Fraldas TURMA DA MONICA Media Pacote...       60.98   
37      326712  Kit com 2 Fraldas TURMA DA MONICA Mega Media P...       76.78   
38      326733  Kit com 2 Fraldas TURMA DA MONICA Mega Media P...       76.78   
39      326731  Kit com 2 Fraldas TURMA DA MONICA Jumbo Grande...       60.98   
40      326734  Kit com 2 Fraldas TURMA DA MONICA Jumbo Grande...       60.98   
41      326715  Kit com 2 Fraldas TURMA DA MONICA Pequena Paco...       60.98   
42      326735  Kit com 2 Fraldas TURMA DA MONICA Pequena Paco...       60.98   
43      326716  Kit com 2 Fraldas TURMA DA MONICA Extra Grande...       60.98   
44      326736  Kit com 2 Fraldas TURMA DA MONICA Extra Grande...       60.98   
45      326717  Kit com 2 Fraldas TURMA DA MONICA Grande Pacot...       76.78   
46      326737  Kit com 2 Fraldas TURMA DA MONICA Grande Pacot...       76.78   
47      326721  Kit com 3 Fraldas TURMA DA MONICA Media Pacote...       91.47   
48      326739  Kit com 3 Fraldas TURMA DA MONICA Media Pacote...       91.47   
49      326740  Kit com 3 Fraldas 

### <strong>Quais são os 10 produtos mais caros da empresa?</strong>

 > R: Os 10 produtos mais caros da empresa estão listados na tabela abaixo


In [15]:
print('Esses são os 10 produtos mais caros da empresa: ')
info_produto = productInformation[['PRODUCT_NAME', 'PRODUCT_VAL']].sort_values('PRODUCT_VAL', ascending=False).head(10)
info_produto = info_produto.rename(columns={'PRODUCT_NAME':'NOME DO PRODUTO','PRODUCT_VAL':'VALOR DO PRODUTO'})
info_produto

Esses são os 10 produtos mais caros da empresa: 


NOME DO PRODUTO  VALOR DO PRODUTO
7942  Whisky Escoces THE MACALLAN Ruby Garrafa 700ml...            741.99
5374  Whisky Escoces JOHNNIE WALKER Blue Label Garra...            735.90
9032   Cafeteira Expresso 3 CORACOES Tres Modo Vermelho            499.00
3057  Vinho Portugues Tinto Vintage QUINTA DO CRASTO...            445.90
9237  Escova Dental Eletrica ORAL B D34 Professional...            399.90
5677  Champagne Rose VEUVE CLICQUOT PONSARDIM Garraf...            366.90
4640  Champagne Frances Brut Imperial MOET Rose Garr...            359.90
8647  Conjunto de Panelas Allegra em Inox TRAMONTINA...            359.00
4449  Whisky Escoces CHIVAS REGAL 18 Anos Garrafa 750ml            329.90
4669  Champagne Frances Brut Imperial MOET & CHANDON...            315.90

### <strong>Quais são as seções dos departamentos 'BEBIDAS' e 'PADARIA'?</strong>

In [16]:
productInformation.head()

PRODUCT_COD                                       PRODUCT_NAME  \
0           10  Acido Tranexamico 250mg Generico EMS 12 Compri...   
1           11  Bissulfato de Clopidogrel 75mg Generico Teuto ...   
2           12  Cloridrato de Amiodarona 200mg Generico Biosin...   
3           13  Acido Tranexâmico 250mg Generico Legrand 12 Co...   
4           14  Cloridrato Oximetazolina Adulto 5mg/ml Genéric...   

   PRODUCT_VAL                DEP_NAME  DEP_COD         SECTION_NAME  \
0        36.71  MEDICAMENTOS GENÉRICOS       10           CIRCULAÇÃO   
1        66.51  MEDICAMENTOS GENÉRICOS       10           CIRCULAÇÃO   
2        31.17  MEDICAMENTOS GENÉRICOS       10           CIRCULAÇÃO   
3        36.71  MEDICAMENTOS GENÉRICOS       10           CIRCULAÇÃO   
4        12.08  MEDICAMENTOS GENÉRICOS       10  GRIPES E RESFRIADOS   

   SECTION_COD  
0           40  
1           40  
2           40  
3           40  
4           41

In [17]:
departamentoPadaria = productInformation.loc[productInformation['DEP_NAME'] == 'PADARIA']
departamentoBebidas = productInformation.loc[productInformation['DEP_NAME'] == 'BEBIDAS']

print('Essas são as seções do \ndepartamento de padaria: ',pd.unique(departamentoPadaria['SECTION_NAME']))
print('\nEssas são as seções do \ndepartamento de bebidas: ',pd.unique(departamentoBebidas['SECTION_NAME']))

Essas são as seções do 
departamento de padaria:  ['DOCES-E-SOBREMESAS' 'QUEIJOS-E-FRIOS' 'PADARIA' 'GESTANTE']

Essas são as seções do 
departamento de bebidas:  ['BEBIDAS' 'VINHOS' 'CERVEJAS' 'REFRESCOS']


### <strong>Qual loja vendeu <span class="girk"><span class="mark">mais</span></span> produtos em um dia? Qual dia?</strong>

 > R: A empresa de código <strong>2</strong> foi a que mais vendeu no dia <strong>03-07-2019</strong>, um total de <strong>24332</strong> produtos.
 

In [18]:
dataStoreSales.head()

STORE_CODE        DATE  SALES_VALUE  SALES_QTY
0           1  2014-11-23    111387.87       7227
1          10  2014-11-23     63256.35       4362
2          11  2014-11-23     62680.86       4257
3          12  2014-11-23     62680.86       4257
4          13  2014-11-23    111387.87       7227

In [19]:
dataStoreSales.loc[dataStoreSales['SALES_QTY'] == dataStoreSales['SALES_QTY'].max()]

STORE_CODE        DATE  SALES_VALUE  SALES_QTY
32133           2  2019-07-03    377886.88      24332

### <strong><span class="girk">Bônus</span>!! Qual foi a venda total de produtos (em $) de cada área de negócio no primeiro trimestre de 2019?</strong>

In [20]:
dataStoreSales.head(1)

STORE_CODE        DATE  SALES_VALUE  SALES_QTY
0           1  2014-11-23    111387.87       7227

In [21]:
dataStoreSales['DATE'] = pd.to_datetime(dataStoreSales['DATE'])
dataStoreSales.head(1)

STORE_CODE       DATE  SALES_VALUE  SALES_QTY
0           1 2014-11-23    111387.87       7227

In [22]:
dataStoreSales.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 35762 entries, 0 to 35761
Data columns (total 4 columns):
 #   Column       Non-Null Count  Dtype         
---  ------       --------------  -----         
 0   STORE_CODE   35762 non-null  int64         
 1   DATE         35762 non-null  datetime64[ns]
 2   SALES_VALUE  35762 non-null  float64       
 3   SALES_QTY    35762 non-null  int64         
dtypes: datetime64[ns](1), float64(1), int64(2)
memory usage: 1.1 MB


In [23]:
dataStoreSales['TRIMESTRE'] = dataStoreSales['DATE'].dt.quarter
dataStoreSales.head(2)

STORE_CODE       DATE  SALES_VALUE  SALES_QTY  TRIMESTRE
0           1 2014-11-23    111387.87       7227          4
1          10 2014-11-23     63256.35       4362          4

In [24]:
ano2019 = dataStoreSales.loc[(dataStoreSales['DATE'].dt.year == 2019) & (dataStoreSales['TRIMESTRE'] == 1)]

In [25]:
ano2019.head(2)

STORE_CODE       DATE  SALES_VALUE  SALES_QTY  TRIMESTRE
28462           1 2019-01-01    196623.22      12838          1
28463          10 2019-01-01    126795.44       4933          1

In [26]:
totalPrimeiroTrimestre = ano2019[['STORE_CODE', 'SALES_VALUE']].groupby('STORE_CODE').sum().reset_index()
totalPrimeiroTrimestre.head()

STORE_CODE  SALES_VALUE
0           1  20096221.15
1           2  20743684.20
2           3  20096221.15
3           4  20096221.15
4           5  20096221.15

In [27]:
dataStoretInfo.head(1)

STORE_CODE STORE_NAME  START_DATE  END_DATA BUSINESS_NAME  BUSINESS_CODE
0           1  Sao Paulo  2006-10-01       NaN        Varejo              1

In [28]:
totalPrimeiroTrimestre['BUSINESS_NAME'] = dataStoretInfo['BUSINESS_NAME']
totalPrimeiroTrimestre.head()

STORE_CODE  SALES_VALUE BUSINESS_NAME
0           1  20096221.15        Varejo
1           2  20743684.20        Varejo
2           3  20096221.15        Varejo
3           4  20096221.15        Varejo
4           5  20096221.15   Proximidade

In [29]:
totalPorArea = totalPrimeiroTrimestre[['BUSINESS_NAME', 'SALES_VALUE']].groupby('BUSINESS_NAME').sum().reset_index()

In [30]:
totalPorArea = totalPorArea.rename(columns = {'BUSINESS_NAME': 'ÁREA RESPONSÁVEL', 
                                              'SALES_VALUE': 'VALOR TOTAL DE VENDAS'}) 

In [31]:
totalPorArea.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5 entries, 0 to 4
Data columns (total 2 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   ÁREA RESPONSÁVEL       5 non-null      object 
 1   VALOR TOTAL DE VENDAS  5 non-null      float64
dtypes: float64(1), object(1)
memory usage: 208.0+ bytes


In [58]:
import plotly.express as px
fig = px.bar(totalPorArea, 
             x='ÁREA RESPONSÁVEL', 
             y='VALOR TOTAL DE VENDAS', 
             title="Vendas 1° Trismestre de 2019",
             color=totalPorArea['ÁREA RESPONSÁVEL'],
             height=600, 
             width=900
            )
fig.show()

In [59]:
# import plotly.io as pio

# pio.write_html(fig, file='_img/vendasprimTri2019.html')
# pio.write_image(fig, file='_img/vendasprimTri2019.svg')


# Construindo visualizaçõ IMDB_movies

## Análise

In [33]:
data = IMDB_movies
data.head()

id                    title                     genre  \
0   1  Guardians of the Galaxy   Action,Adventure,Sci-Fi   
1   2               Prometheus  Adventure,Mystery,Sci-Fi   
2   3                    Split           Horror,Thriller   
3   4                     Sing   Animation,Comedy,Family   
4   5            Suicide Squad  Action,Adventure,Fantasy   

               director                                             actors  \
0            James Gunn  Chris Pratt, Vin Diesel, Bradley Cooper, Zoe S...   
1          Ridley Scott  Noomi Rapace, Logan Marshall-Green, Michael Fa...   
2    M. Night Shyamalan  James McAvoy, Anya Taylor-Joy, Haley Lu Richar...   
3  Christophe Lourdelet  Matthew McConaughey,Reese Witherspoon, Seth Ma...   
4            David Ayer  Will Smith, Jared Leto, Margot Robbie, Viola D...   

   year  runtime  rating   votes  revenueMillions  metascore  
0  2014      121       8  757074            333.0       76.0  
1  2012      124       7  485820            126.0       65.0  
2  2016      117       7  157606            138.0       62.0  
3  2016      108       7   60545            270.0       59.0  
4  2016      123       6  393727            325.0       40.0

In [34]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   id               1000 non-null   int64  
 1   title            1000 non-null   object 
 2   genre            1000 non-null   object 
 3   director         1000 non-null   object 
 4   actors           1000 non-null   object 
 5   year             1000 non-null   int64  
 6   runtime          1000 non-null   int64  
 7   rating           1000 non-null   int64  
 8   votes            1000 non-null   int64  
 9   revenueMillions  872 non-null    float64
 10  metascore        936 non-null    float64
dtypes: float64(2), int64(5), object(4)
memory usage: 86.1+ KB


In [35]:
# colunas com valores nulos
data.isnull().sum()

id                   0
title                0
genre                0
director             0
actors               0
year                 0
runtime              0
rating               0
votes                0
revenueMillions    128
metascore           64
dtype: int64

In [36]:
# valores duplicados ? 
data.duplicated().sum()

0

## Filmes a partir de 2016

In [37]:
print(f"Ano mais recente: {data['year'].max()}")
print(f"Ano mais antigo: {data['year'].min()}")

Ano mais recente: 2016
Ano mais antigo: 2006


Filtrando apenas o ano de 2016

In [38]:
data2016 = data.loc[data['year'] == 2016]
data2016.shape

(297, 11)

In [39]:
# 5 filmes mais votados de 2016
filmesMaisVotados = data2016[['title','votes']].sort_values('votes', ascending=False).head()
primeiroGrafico = filmesMaisVotados
primeiroGrafico = primeiroGrafico.sort_values('votes')

In [40]:
df = data[['director','rating']].groupby('director').mean().reset_index().sort_values('rating', ascending=False)
df.loc[df['rating'] == 9]

director  rating
0                          Aamir Khan     9.0
392                    Makoto Shinkai     9.0
465                     Nitesh Tiwari     9.0
470                   Olivier Nakache     9.0
108                 Christopher Nolan     9.0
194  Florian Henckel von Donnersmarck     9.0

In [41]:
# O diretor que mais publicou filmes
data[['title','director']].groupby('director').count().reset_index().sort_values('title',ascending=False).head()

director  title
518        Ridley Scott      8
391  M. Night Shyamalan      6
155         David Yates      6
422         Michael Bay      6
488  Paul W.S. Anderson      6

In [42]:
# O diretor com filmes mais votados
data[['director','title','votes','rating','year']].sort_values('votes',ascending=False).head()

director                  title    votes  rating  year
54   Christopher Nolan        The Dark Knight  1791916       9  2008
80   Christopher Nolan              Inception  1583625       9  2010
124  Christopher Nolan  The Dark Knight Rises  1222645       9  2012
36   Christopher Nolan           Interstellar  1047747       9  2014
76         Joss Whedon           The Avengers  1045588       8  2012

In [43]:
# Comparando o histórico de votos do diretor que mais publicou filmes com mais votado 
df1 = data[['director','title','votes','rating','year','genre']].loc[data['director'] == 'Christopher Nolan']
df2 = data[['director','title','votes','rating','year','genre']].loc[data['director'] == 'Ridley Scott'] 
segundoGrafico = pd.concat([df1,df2])
segundoGrafico

director                   title    votes  rating  year  \
36   Christopher Nolan            Interstellar  1047747       9  2014   
54   Christopher Nolan         The Dark Knight  1791916       9  2008   
64   Christopher Nolan            The Prestige   913152       9  2006   
80   Christopher Nolan               Inception  1583625       9  2010   
124  Christopher Nolan   The Dark Knight Rises  1222645       9  2012   
1         Ridley Scott              Prometheus   485820       7  2012   
102       Ridley Scott             The Martian   556097       8  2015   
387       Ridley Scott              Robin Hood   221117       7  2010   
470       Ridley Scott       American Gangster   337835       8  2007   
516       Ridley Scott  Exodus: Gods and Kings   137299       6  2014   
521       Ridley Scott           The Counselor    84927       5  2013   
530       Ridley Scott             A Good Year    74674       7  2006   
737       Ridley Scott            Body of Lies   182305       7  2008   

                        genre  
36     Adventure,Drama,Sci-Fi  
54         Action,Crime,Drama  
64       Drama,Mystery,Sci-Fi  
80    Action,Adventure,Sci-Fi  
124           Action,Thriller  
1    Adventure,Mystery,Sci-Fi  
102    Adventure,Drama,Sci-Fi  
387    Action,Adventure,Drama  
470     Biography,Crime,Drama  
516    Action,Adventure,Drama  
521      Crime,Drama,Thriller  
530      Comedy,Drama,Romance  
737      Action,Drama,Romance

## Gráficos

#### Gráfico 1 : Porque escolhi esse gráfico : 

> Queria saber qual foi o filme mais votado. <br>Se fosse uma competição, o filme Deadpool ganharia disparado com pouco mais de 600 mil votos.

In [52]:
import plotly.express as px

fig = px.bar(primeiroGrafico.sort_values('votes'), x='votes',y='title',
             orientation='h', 
             title='Os 5 Filmes Mais Votados de 2016',
             color=[1,2,3,4,5], 
             height=600, 
             width=900)
fig.show()

In [53]:
# import plotly.io as pio

# pio.write_html(fig, file='_img/Os 5 Filmes Mais Votados de 2016.html')
# pio.write_image(fig, file='_img/Os 5 Filmes Mais Votados de 2016.svg')


#### Gráfico 2 : Porque escolhi esse gráfico : 
>Queria visualizar qual foi o diretor que mais publicou filmes e se isso garantia uma boa avaliação e votos ao longo do tempo.<br>
>A partir dessa visualização, vejo que mesmo o diretor Ridley Scott tenha publicado mais filmes ao longo dos anos, não alcançou um número de votos e de avaliações consideráveis quando comparado com o diretor Christopher Nolan

In [54]:
import plotly.express as px
df = px.data.tips()
fig = px.bar(segundoGrafico, x="year", y="votes",
             title='Diretor Mais Votado x Diretor Com Mais Filmes',
             color='director',hover_data=["rating", "title"],
             height=600, 
             width=900)
fig.show()

In [55]:
# import plotly.io as pio

# pio.write_html(fig, file='_img/Diretor Mais Votado x Diretor Com Mais Filmes.html')
# pio.write_image(fig, file='_img/Diretor Mais Votado x Diretor Com Mais Filmes.svg')


# Considerações da última revisão :
>Salvei os gráficos de visualização deste notebook na pasta "_img" para garantir que consigam visualizar as imagens.<br>As imagens de extensão .html, devem apresentar informações hover, enquanto as svg não irão apresentar esta informação.